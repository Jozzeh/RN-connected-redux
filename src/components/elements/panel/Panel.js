import React from 'react';
import {View} from 'react-native';

import {styles} from './Panel.style';

const Panel = ({children}) => {
  return <View style={styles.panel}>{children}</View>;
};

export default Panel;
