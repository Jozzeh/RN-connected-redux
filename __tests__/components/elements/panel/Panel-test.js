import React from 'react';
import {Text} from 'react-native';
import {render} from '@testing-library/react-native';

import Panel from '../../../../src/components/elements/panel/Panel';

/* REDUX */
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

const middlewares = [thunk]; // add your middlewares like `redux-thunk`
const mockStore = configureStore(middlewares);

describe('Panel element', () => {
  let store;

  beforeEach(() => {
    store = mockStore({});
    store.dispatch = jest.fn();
  });

  /* BASIC TEST IF IT RENDERS */
  it('renders', () => {
    render(
      <Provider store={store}>
        <Panel>
          <Text>Renders</Text>
        </Panel>
      </Provider>,
    );
  });

  /* CREATE SNAPSHOTS ON ELEMENTS THAT DO NOT CHANGE OFTEN */
  it('matches snapshot', () => {
    const {toJSON} = render(
      <Provider store={store}>
        <Panel>
          <Text>Renders</Text>
        </Panel>
      </Provider>,
    );
    expect(toJSON).toMatchSnapshot();
  });
});
