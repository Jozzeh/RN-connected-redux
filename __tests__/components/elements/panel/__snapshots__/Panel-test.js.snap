// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Panel element matches snapshot 1`] = `[Function]`;
